File: Horas.cpp
//Archivo Hora.cpp

#include "Horas.h"

Horas::Horas(int hora, int min) : Hora(hora), Min(min){}
	

Horas::Horas(){
	Hora = 6;
	Min = 0;
}
void Horas::Show_Hora(){
	cout << Hora << ":" << Min << endl;
}
void Horas::SetHora(int hora, int min){
	Hora = hora;
	Min = min;
}

Horas Horas::GetHora(){
	return *this;
}

int Horas::Get_H(){
	return Hora;
}

int Horas::Get_M(){
	return Min;
}

void Horas::Add_Min(){
	Min += 10;
	if(Min >= 60){
		Hora += 1;
		Min = 0;
		if(Hora >= 24){
			Hora = 0;
		}
	}
}

bool Horas::operator>=(Horas hora){
    cout << "Hora: " << Hora << ", Hora de referencia: " << hora.Get_H() << endl;
    cout << "Min: " << Min << ", Min de referencia: " << hora.Get_M() << endl;
    return (Hora >= hora.Get_H()) && (Min >= hora.Get_M());
}




bool Horas::operator==(Horas hora){
    return (Hora == hora.Get_H()) && (Min == hora.Get_M());
}

File: proyectoAeropuertoPython/Horas.py
class Horas:
    def __init__(self, hora=6, minuto=0):
        self.Hora = hora
        self.Min = minuto

    def Show_Hora(self):
        print(f"{self.Hora:02d}:{self.Min:02d}")

    def SetHora(self, hora, minuto):
        self.Hora = hora
        self.Min = minuto

    def GetHora(self):
        return self

    def Get_H(self):
        return self.Hora

    def Get_M(self):
        return self.Min

    def Add_Min(self):
        self.Min += 10
        if self.Min >= 60:
            self.Hora += 1
            self.Min = 0
            if self.Hora >= 24:
                self.Hora = 0

    def __ge__(self, hora):
        return (self.Hora, self.Min) >= (hora.Get_H(), hora.Get_M())

    def __eq__(self, hora):
        return (self.Hora, self.Min) == (hora.Get_H(), hora.Get_M())
